### ISSUE TITLE
Multi region support in same cluster

### ISSUE THREAD

**Issue by dcarrion87** [aws/karpenter#5024]
### Description

**What problem are you trying to solve?**

We would like to extend one karpenter to handling multi region support in a single Kubernetes cluster.

I can see subnetSelector assumes the current region.

Would we need to deploy multiple Karpenters to handle this?

**Comment by jonathan-innis** [comment]
We don't currently have multi-region support. The biggest reason for this is the expected communication latency between the worker nodes and the control-plane, which I'm assuming has to be a single region, even in k3s due to etcd leader being tied to a single region. 

Is there a way that you overcome that latency?

**Comment by dcarrion87** [comment]
For our use case it's not an issue. 140ms is totally fine for where these worker nodes are going to be doing.

Is it possible to run multiple Karpenters in the same cluster with AWS_REGION set and monitoring different provisioners / node templates? Or will they trip over each other?

**Comment by dcarrion87** [comment]
> We don't currently have multi-region support. The biggest reason for this is the expected communication latency between the worker nodes and the control-plane, which I'm assuming has to be a single region, even in k3s due to etcd leader being tied to a single region.
> 
> Is there a way that you overcome that latency?

@jonathan-innis For our use case it's not an issue. 140ms is totally fine for where these worker nodes are going to be doing.

Is it possible to run multiple Karpenters in the same cluster with AWS_REGION set differently and monitoring different provisioners / node templates? Or will they trip over each other?

**Comment by jonathan-innis** [comment]
> Or will they trip over each other

Yeah, they'd definitely trip over each other. There's potentially a possibility that sometime deep in the future we might support the ability to run two at one time in the same cluster with some kind of global lock/lease hand-off mechanism, but right now, they are running with the assumption that they are a singleton in the cluster.

**Comment by dcarrion87** [comment]
Thank you. Appreciate the engagement.

**Comment by jonathan-innis** [comment]
> Appreciate the discussion and engagement

No problem 👍 Definitely think that this could be a neat feature down-the-line, whether we support multi-region Karpenter natively or whether we allow multiple Karpenters to run in the same cluster

**Comment by montanaflynn** [comment]
Jumping in here with the same request. Our use case involves running large batch inference jobs where AWS can run out of GPUs in a single region. We want to have our single cluster be able to provision nodes from multiple regions.

**Comment by jukops** [comment]
Instances such as GPU or Spot servers may not be available in a single region. Even though EKS Auto Mode is supported, adding new EKS clusters is still not efficient in terms of management and cost. If we could use that feature, it would be a good method for us to use EKS clusters more efficiently.
